WordPress/Classes/ViewRelated/Reader/Tab Navigation/ReaderTabItemsStore.swift

```diff
@@ -86,14 +86,35 @@ extension ReaderTabItemsStore {
             return
         }
         state = .loading
-        service.fetchReaderMenu(success: { [weak self] in
-                                    self?.fetchTabBarItems()
-            },
-                                failure: { [weak self] error in
-                                    let actualError = error ?? NSError(domain: WordPressComRestApiErrorDomain, code: -1, userInfo: nil)
-                                    DDLogError(ReaderTopicsConstants.remoteFetchError + actualError.localizedDescription)
-                                    self?.fetchTabBarItems()
+
+        let dispatchGroup = DispatchGroup()
+
+        // Sync the reader menu
+        dispatchGroup.enter()
+        service.fetchReaderMenu(success: {
+            dispatchGroup.leave()
+        }, failure: { (error) in
+            let actualError = error ?? NSError(domain: WordPressComRestApiErrorDomain, code: -1, userInfo: nil)
```

Comment:
This needed fixing before your changes: can we use ReaderTopicsConstants.remoteServiceError instead of creating the default error here and in `fetchFollowedSites` below?

Reply:
Done in: https://github.com/wordpress-mobile/WordPress-iOS/pull/14370/commits/7c7ff3010eeb891704b011d52c6f1d8901afe380